SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -599,23 +598,36 @@ void USpatialActorChannel::RegisterEntityId(const FEntityId& ActorEntityId)
 	}
 }
 
-void USpatialActorChannel::OnCreateEntityResponse(const worker::CreateEntityResponseOp& Op)
+void USpatialActorChannel::OnReserveEntityIdResponse(const Worker_ReserveEntityIdResponseOp& Op)
 {
-	check(SpatialNetDriver->GetNetMode() < NM_Client);
+	if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
+	{
+		UE_LOG(LogSpatialGDKActorChannel, Error, TEXT("Failed to reserve entity id. Reason: %s"), UTF8_TO_TCHAR(Op.message));
+		// TODO: From now on, this actor channel will be useless. We need better error handling, or a retry mechanism here.
+		return;
+	}
+	UE_LOG(LogSpatialGDKActorChannel, Log, TEXT("Received entity id (%lld) for: %s."), Op.entity_id, *Actor->GetName());
+	EntityId = Op.entity_id;
```

Comment:
We should pass `Op.entity_id` directly into RegisterEntityId, and store it in EntityId there.